TestingGrounds/TestingGrounds/HashTable.cpp

```diff
@@ -63,26 +60,80 @@ uint32_t HashTable<K, V>::calcPrime( void )
     return ret;
 }
 
-template<uint32_t K, uint32_t V>
-bool HashTable<K, V>::keyExists( void )
+bool HashTable::keyExists( uint32_t key )
 {
+    if(size == 0)
+        return false;
+    else if(size == 1)
+        return table[0]->getValue() == key;
 
+    for(int i = 0; i < size; i++)
+    {
+        if(table[i]->getValue() == key)
+            return true;
+    }
 }
 
-template<uint32_t K, uint32_t V>
-uint32_t HashTable<K, V>::getValue(void)
+uint32_t HashTable::getValue( uint32_t key )
 {
+    uint32_t hash1 = hashFunction(key);
+    uint32_t hash2 = doubleHashFunction(key);
 
+    HashNode *entry = table[hash1];
+
+    if(entry->getKey() == key)
+        return entry->getValue();
+    else
+    {
+        HashNode* offsetEntry = table[hash2];
+        return offsetEntry->getValue();
+    }
 }
 
-template<uint32_t K, uint32_t V>
-void HashTable<K, V>::putPair( void )
+void HashTable::put( uint32_t key, uint32_t value )
 {
+    HashNode* newNode = new HashNode(key, value);
 
+    uint32_t hash1 = hashFunction(key);
+    uint32_t hash2 = doubleHashFunction(key);
+
+    if(table[hash1] == nullptr)
+        table[hash1] = newNode;
+    else if(table[hash2] == nullptr)
+        table[hash2] == nullptr;
+    else
+        return;
+    size++;
 }
 
-template<uint32_t K, uint32_t V>
-void HashTable<K, V>::remove( void )
+void HashTable::remove( uint32_t key )
 {
+    uint32_t hash1 = hashFunction(key);
+    uint32_t hash2 = doubleHashFunction(key);
+
+    HashNode* temp = table[hash1];
+    HashNode* doubleTemp = table[hash2];
 
+    if(temp->getKey() == key)
+    {
+        delete temp;
+        table[hash1] = nullptr;
+    }
+    else if(doubleTemp->getKey() == key)
+    {
+        delete doubleTemp;
+        table[hash2] = nullptr;
+    }
+    else
+        return;
+    size--;
+}
+
+void HashTable::printTable( void )
+{
+    for(int i = 0; i < capacity; i++)
```

Comment:
Mixing of signed/unsigned integers in the method.

Reply:
unsigned!